library/src/compute/calc.rs

```diff
@@ -131,6 +131,43 @@ pub fn pow(
     result.value()
 }
 
+/// Raise a value to some exponent of e.
+///
+/// ## Example { #example }
+/// ```example
+/// #calc.exp(3)
+/// ```
+///
+/// Display: Power
+/// Category: calculate
+/// Returns: integer or float
+#[func]
+pub fn exp(
+    /// The exponent of the power. Must be non-negative.
+    exponent: Spanned<Num>,
+) -> Value {
+    match exponent.v {
+        Num::Int(i) if i32::try_from(i).is_err() => {
+            bail!(exponent.span, "exponent is too large")
+        }
+        Num::Float(f) if !f.is_normal() && f != 0.0 => {
+            bail!(exponent.span, "exponent may not be infinite, subnormal, or NaN")
+        }
+        _ => {}
+    };
+
+    let result = match exponent.v{
+        Num::Int(i) => Num::Float((i as f64).exp()),
+        Num::Float(f) => Num::Float(f.exp()),
+    };
+
+    if result.float().is_nan() {
+        bail!(args.span, "the result is not a real number")
+    }
```

Comment:
It can't be triggered since that you only accept real numbers. It'll always return real numbers.

Reply:
@HarmoGlace why would it be different from normal `pow`? It also has that check.